data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/Cne.java

```diff
@@ -94,6 +98,64 @@ private void addTimeSeriesToCne(DateTime networkDate) {
     private void createAllConstraintSeries(Point point) {
 
-        point.constraintSeries = new ArrayList<>();
+        Crac crac = cneHelper.getCrac();
+        Network network = cneHelper.getNetwork();
+
+        List<ConstraintSeries> constraintSeriesList = new ArrayList<>();
+
+        crac.getCnecs().forEach(cnec -> {
+            /* Country of cnecs */
+            Line cnecLine = network.getLine(cnec.getNetworkElement().getId());
+            Set<Country> countries = new HashSet<>();
+            // check if the cnec is cross zonal
+            if (cnecLine != null) {
+                cnecLine.getTerminal1().getVoltageLevel().getSubstation().getCountry().ifPresent(countries::add);
+                cnecLine.getTerminal2().getVoltageLevel().getSubstation().getCountry().ifPresent(countries::add);
+            }
+
+            /* Create Constraint series */
+            ConstraintSeries constraintSeriesB54 = newConstraintSeries(cnec.getId(), B54_BUSINESS_TYPE, countries, OPTIMIZED_MARKET_STATUS);
+            ConstraintSeries constraintSeriesB57 = newConstraintSeries(cnec.getId(), B57_BUSINESS_TYPE, countries, OPTIMIZED_MARKET_STATUS);
+            ConstraintSeries constraintSeriesB88 = newConstraintSeries(cnec.getId(), B88_BUSINESS_TYPE, countries, OPTIMIZED_MARKET_STATUS);
+
+            /* Add contingency if exists */
+            Optional<Contingency> optionalContingency = cnec.getState().getContingency();
+            if (optionalContingency.isPresent()) {
+                ContingencySeries contingencySeries = newContingencySeries(optionalContingency.get().getId(), optionalContingency.get().getName());
+                constraintSeriesB54.contingencySeries.add(contingencySeries);
+                constraintSeriesB57.contingencySeries.add(contingencySeries);
+                constraintSeriesB88.contingencySeries.add(contingencySeries);
+            }
+
+            /* Add critical network element */
+            List<Analog> measurementsB54 = new ArrayList<>();
+            List<Analog> measurementsB57 = new ArrayList<>();
+            List<Analog> measurementsB88 = new ArrayList<>();
+
+            CnecResultExtension cnecResultExtension = cnec.getExtension(CnecResultExtension.class);
+            if (cnecResultExtension != null) {
+                String measurementType = cneHelper.instantToCodeConverter(cnec.getState().getInstant());
+                CneCnecsCreator.createB54B57Measurements(cnec, measurementType, cneHelper.getPostOptimVariantId(), measurementsB54, measurementsB57);
+                CneCnecsCreator.createB88Measurements(cnec, measurementType, cneHelper.getPreOptimVariantId(), measurementsB88);
+
+                MonitoredRegisteredResource monitoredRegisteredResourceB54 = newMonitoredRegisteredResource(cnec.getId(), cnec.getName(), findNodeInNetwork(cnec.getNetworkElement().getId(), network, Branch.Side.ONE), findNodeInNetwork(cnec.getNetworkElement().getId(), network, Branch.Side.TWO), measurementsB54);
```

Comment:
You can pass 3 arguments to this method: cnec, network and measurementsB54. It will make this part of the code more readable